packages/core/src/FindContractVersion.js

```diff
@@ -1,4 +1,13 @@
 const assert = require("assert");
+const path = require("path");
+const fs = require("fs");
+
+let latestVersionMap = {};
+try {
+  latestVersionMap = JSON.parse(fs.readFileSync(`${path.resolve(__dirname)}/../build/contract-type-hash-map.json`));
+} catch (error) {
+  console.log("WARNING: latest version map was not found in the build directory! Run `yarn build` from core first!");
+}
```

Comment:
Are we expecting this to be used in UIs? Is this function exported in browser.js?

If we are expecting to use it in UIs (we might not be!), we might need to come up with a browser-safe way of branching around this. I think doing the require in the path and fs requires inside the try catch would work. You could also branch on the browser env specifically too.

Reply:
the only place this is used (right now) is in the bot's index.js files when initializing the type of contract. you are right that this is not browser compatible and we totally would need to solve that. However, I think it might be fine to just solve our CI issues for now with this as there are going to be a _lot_ of other browser-specific things that might not work too well in the `core` package anyways.

Reply:
Okay, sounds good. We'll tackle it when we get there.